src/one_dragon/utils/gpu_executor.py

```
@@ -1,17 +1,55 @@
-    _executor.shutdown(wait=wait)
\ No newline at end of file
+def run_sync(fn: Callable[..., T], /, *args, **kwargs) -> T:
+    if is_executor_thread():
+        return fn(*args, **kwargs)
+    return submit(fn, *args, **kwargs).result()
+
+
+def should_serialize_session(session) -> bool:
+    try:
+        providers = session.get_providers()
+    except Exception:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)